llvm/test/CodeGen/AArch64/abd-combine.ll

```diff
@@ -93,9 +147,14 @@ define <8 x i16> @abdu_const_bothhigh() {
 }
 
 define <8 x i16> @abdu_undef(<8 x i16> %src1) {
-; CHECK-LABEL: abdu_undef:
-; CHECK:       // %bb.0:
-; CHECK-NEXT:    ret
+; CHECK-SD-LABEL: abdu_undef:
+; CHECK-SD:       // %bb.0:
+; CHECK-SD-NEXT:    ret
+;
+; CHECK-GI-LABEL: abdu_undef:
+; CHECK-GI:       // %bb.0:
+; CHECK-GI-NEXT:    uabd v0.8h, v0.8h, v0.8h
```

Comment:
Common problem.

Reply:
It is again the trunc of abs to abdu combine, but there is an undef hidden in the LLVM-IR.

Reply:
The undef is on the RHS.

Reply:
There's no problem. The undef should have already propagated to fold the whole result to undef in the original IR. You could fold undef inputs to the new node if you really wanted this to disappear

Reply:
`%zextsrc2 = zext <8 x i16> undef to <8 x i32>`. We currently don't fold G_ZEXT(undef). We end up with G_ABDU(%src1, undef), which might explain the v0.

Reply:
undef -> undef or undef -> zero, this is only a missed optimization (that should have already folded in the IR). It's not wrong or a problem for this combine to deal with

Reply:
The IRTranslator almost faithfully imports the LLVM-IR. We combine the pattern into G_ABDU(%src1, undef). The instruction selector selects the G_ABDU as an uabd. My guess is that undef lead to picking the default register for the RHS aka v0. It looks odd that both operands are v0, but the RHS is probably the default pick.

Reply:
Undef can produce any register input, and will almost always end up using the first register in the allocation order